frontend/src/hooks/queryBuilder/useQueryOperations.ts

```diff
@@ -0,0 +1,294 @@
+	handleChangeDataSource: (newSource: DataSource) => void;
+	handleToggleDisableQuery: () => void;
+	handleChangeGroupByKeys: (values: BaseAutocompleteData[]) => void;
+	handleChangeQueryLegend: (e: ChangeEvent<HTMLInputElement>) => void;
+	handleChangeReduceTo: (value: ReduceOperators) => void;
+	handleChangeHavingFilter: (having: Having[]) => void;
+	handleDeleteQuery: () => void;
+	handleChangeOrderByKeys: (value: BaseAutocompleteData[]) => void;
+	handleChangeLimit: (value: number | null) => void;
+	handleChangeAggregateEvery: (value: number) => void;
+	handleChangeTagFilters: (value: TagFilter) => void;
+};
+
+export const useQueryOperations: UseQueryOperations = ({
+	query,
+	index,
+	panelType,
+}) => {
+	const { handleSetQueryData, removeEntityByIndex } = useQueryBuilderContext();
+	const [operators, setOperators] = useState<string[]>([]);
+
+	const { dataSource } = query;
+
+	const handleChangeOperator = useCallback(
+		(value: string): void => {
+			const aggregateDataType: BaseAutocompleteData['dataType'] =
+				query.aggregateAttribute.dataType;
+
+			const newQuery: IBuilderQueryForm = {
+				...query,
+				aggregateOperator: value,
+				having: [],
+				groupBy: [],
+				orderBy: [],
+				limit: null,
+				tagFilters: { items: [], op: 'AND' },
+			};
+
+			switch (aggregateDataType) {
+				case 'string':
+				case 'bool': {
+					const typeOfValue = findDataTypeOfOperator(value);
+
+					handleSetQueryData(index, {
+						...newQuery,
+						...(typeOfValue === 'number'
+							? { aggregateAttribute: initialAggregateAttribute }
+							: {}),
+					});
+
+					break;
+				}
+				case 'float64':
+				case 'int64': {
+					handleSetQueryData(index, newQuery);
+
+					break;
+				}
+
+				default: {
+					handleSetQueryData(index, newQuery);
+					break;
+				}
+			}
+		},
+		[index, query, handleSetQueryData],
+	);
+
+	const getNewOperators = useCallback(
+		(dataSource: DataSource, currentPanelType: ITEMS): string[] => {
+			let operatorsByDataSource = mapOfOperators[dataSource];
+
+			if (dataSource !== DataSource.METRICS && currentPanelType !== 'list') {
+				operatorsByDataSource = operatorsByDataSource.filter(
+					(operator) => operator !== StringOperators.NOOP,
+				);
+			}
+
+			return operatorsByDataSource;
+		},
+		[],
+	);
+
+	const handleChangeAggregatorAttribute = useCallback(
+		(value: BaseAutocompleteData): void => {
+			const newQuery: IBuilderQueryForm = {
+				...query,
+				aggregateAttribute: value,
+				having: [],
+			};
+
+			handleSetQueryData(index, newQuery);
+		},
+		[index, query, handleSetQueryData],
+	);
+
+	const handleChangeDataSource = useCallback(
+		(nextSource: DataSource): void => {
+			let newQuery: IBuilderQueryForm = {
+				...query,
+				dataSource: nextSource,
+			};
+
+			const newOperators = getNewOperators(nextSource, panelType);
+
+			const initCopy = {
+				...(initialQueryBuilderFormValues as Partial<IBuilderQueryForm>),
+			};
+			delete initCopy.queryName;
```

Comment:
please filter the keys... 

Delete to remove the property from the object because this mutates the original, which can lead to unpredictable behaviours and becomes difficult to debug.